R/ContextEvaluation.R

```diff
@@ -0,0 +1,78 @@
+#' @title Evaluation Context
+#'
+#' @description
+#' A [CallbackEvaluation] accesses and modifies data during [resample()] and [benchmark()] via the `ContextEvaluation`.
+#' See the section on fields for a list of modifiable objects.
+#' See [callback_evaluation()] for a list of stages that access `ContextEvaluation`.
+#'
+#' @export
+ContextEvaluation = R6Class("ContextEvaluation",
+  inherit = Context,
+  public = list(
+
+    #' @field task ([Task])\cr
+    #' The task to be evaluated.
+    #' The task is unchanged during the evaluation.
+    task = NULL,
+
+    #' @field learner ([Learner])\cr
+    #' The learner to be evaluated.
+    #' The learner contains the models after stage `on_evaluation_before_train`.
+    learner = NULL,
+
+    #' @field resampling [Resampling]\cr
+    #' The resampling strategy to be used.
+    #' The resampling is unchanged during the evaluation.
+    resampling = NULL,
+
+    #' @field param_values `list()`\cr
+    #' The parameter values to be used.
+    #' Is usually only set while tuning.
```

Comment:
it this not accessible via the learner anyway?

Reply:
--> I would probably not expose things like this via the context not not restrict ourselves in the future when we want to refactor the workhorse

Reply:
Yes you are right. Should be enough to make this accessible via `$learner` in `on_evaluation_before_train`.